agot-bg-game-server/src/common/ingame-game-state/action-game-state/resolve-march-order-game-state/combat-game-state/CombatGameState.ts

```diff
@@ -501,6 +526,8 @@ export default class CombatGameState extends GameState<
                 return s;
             }
 
+            // This solution seems to fail for Margaery vs Mance, as both cards may overwrite Baratheon's final combat strength
+            // and this implementation will cause the last resolved house card to win instead of the first.
             return houseCard.ability ? houseCard.ability.finalCombatStrength(this, house, houseCard, affectedHouseCard, s) : s;
```

Comment:
@Longwelwind @mmasztalerczuk During the fix of this issue I realized another problem described in the comment. Do you have any smart solution how to end `reduce` in case another house card already overwrote the final combat strength? My approach was quite complex but I guess it's much simpler... 😆

Reply:
Can't you continue the reduce, and after the reduce has finished, check if the strength has been overwritten? Basically, have 2 flows: one for "simple addition/substraction" final combat strength modification, and an other one for "overwritten" final combat strength modification?

It would technically lead to wasted calculation, but it could make the code clearer & simpler

Reply:
It took a while but I guess I understood what you mean. So we have a method `modifyFinalCombatStrength` in `houseCardAbilities` which calculates the difference of the initial combat strength to the specified final combat strength and we apply a possible second final modifier only if no modification has been applied yet?

But if I understood correctly that still might be problematic in case Baratheon already has a final strength of 2. In case Tyrell is in front on Iron Throne it looks like there is no difference and Mance's modification will be applied. But probably I just haven't got it correctly yet :expressionless:

Reply:
Thanks for your hint on Discord. According to https://boardgamegeek.com/thread/867434/final-combat-strength it makes sense to follow the regular rules which means in this special case Mance might benefit from a later Iron Throne position. 

And after reading this forum discussion I come to the conclusion that the current implementation maybe is not really buggy but it might ask for VSB for nothing. Probably right now you might be asked for the bonus and if you apply it the recalculation will remove the bonus because of the final again...